tools/knative_setup.sh

```diff
@@ -0,0 +1,395 @@
+#!/usr/bin/env bash
+
+# Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
Please add the source of this file - where did you take it from? In addition to license, we should specify who is the author.

Did you modify the file or just copy it?